parser_library/src/parsing/grammar/model_operand_rules.g4

```diff
@@ -57,14 +57,21 @@ model_string returns [std::string value]
 		collector.add_hl_symbol(token_info(provider.get_range($ap1,$ap2),hl_scopes::string));
 	};
 
+before_var_sym_model_string returns [std::string value]
+	: ap1=APOSTROPHE model_string_ch_c ap2=(APOSTROPHE|ATTR)
+	{
+	    $value.append(std::move($model_string_ch_c.value));
```

Comment:
does this perform the `if ($l_sp_ch.value == "&&")` replacement? Or is it not needed?

Reply:
No, during the first passthrough it doesn't do the `&&` replacement. Actually, one could see the replacement to be part of the issue.

There are two passthroughs that happen when strings like `AINSERT '&&B    SETC ''&SYSSTMT''',BACK` are processed.

Originally, during the first passthrough `ANTLR` would return the following result `AINSERT '&B    SETC ''00000009''',BACK`. Unfortunately, during the second pass the string would go through the same branch as before which ultimately resulted in error.

The idea behind this modification is that the processing during the second pass should behave similarly to the processing of a simpler string like `AINSERT '&&B    SETC ''00000009''',BACK` which requires only 1 passthrough. That means that only the `&SYSSTMT` part of the original string is modified and the rest stays the same until it go to `ANTLR` for the second time.

Reply:
👍 
I suggest adding additional tests with multiple substitutions and `''` sequences...